packages/builder/src/helpers/WhereQuery.ts

```diff
@@ -1,194 +1,165 @@
-        );
-        return 'discarded';
+        return new Writer('discarded', [{
+            type: 'no searchable',
+            message: 'There are no allowed columns, all columns will be ignored'
+        }]);
     }
     if (column === 'match' && searchableCols.length > 0) {
-        return 'match';
+        return Writer.of('match');
     }
     if (searchableCols.indexOf(column) !== -1) {
-        return 'query';
+        return Writer.of('query');
     }
     if (
         column.indexOf('not_') === 0 &&
         searchableCols.indexOf(column.substr(4)) !== -1
     ) {
-        return 'not';
+        return Writer.of('not');
     }
     if (
         column.indexOf('from_') === 0 &&
         searchableCols.indexOf(column.substr(5)) !== -1
     ) {
-        return 'from';
+        return Writer.of('from');
     }
     if (
         column.indexOf('to_') === 0 &&
         searchableCols.indexOf(column.substr(3)) !== -1
     ) {
-        return 'to';
+        return Writer.of('to');
     }
     if (
         column.indexOf('like_') === 0 &&
         searchableCols.indexOf(column.substr(5)) !== -1
     ) {
-        return 'like';
+        return Writer.of('like');
     }
     if (
         column.indexOf('not_like_') === 0 &&
         searchableCols.indexOf(column.substr(9)) !== -1
     ) {
-        return 'not like';
+        return Writer.of('notLike');
     }
 
-    signale.warn(
-        `Ignoring filter: ${column}. Allowed columns: ${searchableCols}`,
-    );
-
-    return 'discarded';
+    return new Writer('discarded', [{ type: 'ignoring', message: column }]);
 };
 
 export const sortQueryType = (filters, searchableCols) => {
-    return Object.keys(filters).reduce(
-        (result, col) => {
-            const colType = getColType(col, searchableCols);
-
-            return {
-                ...result,
-                [colType]: { ...result[colType], [col]: filters[col] },
-            };
-        },
-        { match: {}, from: {}, to: {}, query: {} },
-    );
+    return new List(Object.keys(filters)).map(col => {
+        const typeWriter = getColType(col, searchableCols);
+
+        return typeWriter.map(type => ({
+            type,
+            col,
+            value: filters[col],
+        }));
+    });
 };
 
-export const getMatch = (filters, searchableCols, whereParts = []) => {
+export const getMatch = (col, value, searchableCols) => {
     return !searchableCols.length
-        ? whereParts
-        : Object.keys(filters).reduce(
-              result => [
-                  ...result,
-                  `(${searchableCols
-                      .map(
-                          searchableCol =>
-                              `${searchableCol}::text ILIKE $match`,
-                      )
-                      .join(' OR ')})`,
-              ],
-              whereParts,
-          );
+        ? null
+        : `(${searchableCols.map(searchableCol => `${searchableCol}::text ILIKE $match`).join(' OR ')})`;
```

Comment:
If you can improve the readability of that line, that would be great.
But not mandatory.